core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/Scheduler.scala

```diff
@@ -289,6 +292,11 @@ object Scheduler {
 
     implicit val logger = new AkkaLogging(akka.event.Logging.getLogger(actorSystem, this))
 
+    if (useClusterBootstrap) {
+      AkkaManagement(actorSystem).start()
+      ClusterBootstrap(actorSystem).start()
+    }
```

Comment:
Refer to the sharding place, maybe we can make it stronger?
![image](https://user-images.githubusercontent.com/11749867/180124549-33228c6b-67a3-4e28-9a8c-3cb7b6428263.png)

Reply:
@ningyougang, I believe the ShardingContainerPoolBalancer only calls Cluster(actorSystem) because it needs access to the cluster object in order to manage cluster state, as part of the main functionality of that class. Since the Scheduler and Controller are not performing cluster management tasks, I do not think they need to make this call (and thus do not need the check for seed nodes). 

If we wanted to do a check just to make sure configuration is correct, I could add that (e.g.,
```scala
...
} else {
   /* create error if seed nodes not set */
  loadConfigOrThrow[Seq[String]]("akka.cluster.seed-nodes")
}
```

What do you think?

Reply:
I am not sure, wWould love to hear other comments for this.

Reply:
The `ShardingContaienrPoolBalancer` considers both seed-nodes-based and discovery-based approaches.
It looks this is to support both ansible-based deployment and deployment on K8S.
https://github.com/apache/openwhisk/pull/3548

I believe discovery would work with Ansible too but it would be great to keep options for those who are using seed nodes.

Reply:
I believe my proposed changes work just fine with seed nodes (in the single-replica controller and scheduler deployments with Kubernetes, I use a single seed node and not cluster discovery to save time on component startup). However, I believe the loadConfigOrThrow proposed above also works with seed nodes set as environment variables - so both solutions should work well with ansible and seed nodes.

The only different for adding the logic pointed out by @ningyougang (e.g., code snippet in my last comment) is that it would fail if useCluster flag is not specified AND seed nodes are not specified. Currently, if neither is specified, there is no explicit failure, a cluster is just not created properly... but the controller (and scheduler) still appear to run without issue in that case.

Reply:
One more thought - the ShardingContainerPoolBalancer does not fail if a cluster is not created properly, so I took that approach here too.

Reply:
> I believe my proposed changes work just fine with seed nodes (in the single-replica controller and scheduler deployments with Kubernetes

This is where I feel we are not on the same page.
In my downstream, we are running controllers and schedulers with more than 1 replica without any issue.
But anyway, I don't want to impede this PR from being merged.